### PR TITLE
ci: Add test script for vfio-ap on s390x

### PR DESCRIPTION
**Pull request by BbolroC**
Add a test script to verify a funtionality of vfio-ap on s390x.
Add Dockerfile for a test image where s390-tools is installed.

Fixes: #5212

Signed-off-by: Hyounggyu Choi <Hyounggyu.Choi@ibm.com>

### CODE REVIEW

File: functional/vfio-ap/run.sh [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+build_test_image() {
+    ${container_engine} rmi -f ${test_image_name}
+    ${container_engine} build -t ${test_image_name} ${script_path}
+    ${container_engine} push ${test_image_name}
+}
+
+create_mediated_device() {
+    # a device lastly listed is chosen
+    APQN=$(lszcrypt | tail -1 | awk '{ print $1}')
+    if [[ ! $APQN =~ [0-9]{2}.[0-9]{4} ]]; then
```
Comment (hbrueckner) [inline functional/vfio-ap/run.sh:109]: The `lszcrypt` output shows card and domain in hex. So the regex should better use: `[[:xdigit:]]{2}.[[:xdigit:]]{4}` here.

File: functional/vfio-ap/run.sh [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+        echo "Incorrect format for APQN" >&2
+        exit 1
+    fi
+    _APID=${APQN//.*}
```
Comment (hbrueckner) [inline functional/vfio-ap/run.sh:113]: minor: You might fold this into the `echo` command following this.

File: functional/vfio-ap/run.sh (hunk 1 above)
Comment (BbolroC) [reply functional/vfio-ap/run.sh:109]: oh, my mistake! thanks for pointing this out.

File: functional/vfio-ap/run.sh (hunk 2 above)
Comment (BbolroC) [reply functional/vfio-ap/run.sh:113]: They are used later at https://github.com/kata-containers/tests/pull/5225/files#diff-63a904a0cb68f75720776899484f88be8037d7d94d404537b84da324c49ab7c2R156 like:

```
[ -n "${dev_index}" ] && sudo ctr run --runtime io.containerd.run.kata.v2 --rm -t --device ${dev_base}/${dev_index} ${test_image_name} test bash -c "lszcrypt ${_APID}.${_APQI}"
```

Comment (hbrueckner) [reply functional/vfio-ap/run.sh:113]: Ok.. then I everything is OK ;-) Thanks.

**Comment by hbrueckner** [comment]
Hi  @BbolroC , this looks nice to me know! Thanks a lot!

/lgtm

**Review by hbrueckner — APPROVED** [review]
LGTM

**Comment by GabyCT** [comment]
are we planning to add a new CI job to test this?

**Comment by BbolroC** [comment]
> are we planning to add a new CI job to test this?

Yes, I am going to create a dedicated job when a machine is ready.

**Review by GabyCT — APPROVED** [review]

**Comment by BbolroC** [comment]
I've created a Jenkins job which fetches and dumps log of the internal test (there is still restriction on getting a public machine ready, to which a VFIO-AP device is attached). The test runs on a daily basis and the job will be triggered accordingly. Here is a result of the first run:

http://jenkins.katacontainers.io/job/kata-containers-2.0-ubuntu-20.04-s390x-VFIO-AP-daily/6/consoleText

ATM, the test runs depending on https://github.com/kata-containers/kata-containers/pull/4582, which should get reviewed sooner or later.

**Review by fidencio — APPROVED** [review]
lgtm, on behalf of @hbrueckner! :-)

**Pull request merged**